ansibullbot/utils/shippable_api.py

```diff
@@ -350,3 +323,60 @@ def get_test_results(self, run_id, usecache=False, filter_paths=[]):
                         break
 
         return (run_data, commitSha, results, ci_verified)
+
+
+    def rebuild(self, run_number):
+        """trigger a new run"""
+        run_url = "%s&runNumbers=%s" % (ANSIBLE_RUNS_URL, run_number)
+        response = self.fetch(run_url)
+        if not response:
+            raise Exception("Unable to fetch %r" % run_url)
+        self.check_response(response)
+
+        run_id = response.json()[0]['id']
+
+        newbuild_url = "%s/projects/%s/newBuild" % (SHIPPABLE_URL, ANSIBLE_PROJECT_ID)
+        response = self.fetch(newbuild_url, verb='post', data=json.dumps({'runId':run_id}))
```

Comment:
We need some kind of logic to protect against repeated rebuilds of PRs. Since this depends on the build status reported by GH, it can get out of sync with what Shippable is doing. This can happen due to lost or delayed web hooks. We should probably be getting the CI status from Shippable itself, instead of relying on GH.